javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -151,67 +169,66 @@ private boolean arePointingToSameLeaf(int i, int j) {
     }
 
     /* drop root node while it has a single element */
-    private static <T> BitMappedTrie<T> collapsed(Object[] array, int offset, int length, int shift) {
+    private static <T> BitMappedTrie<T> collapsed(ArrayType<T> type, Object array, int offset, int length, int shift) {
         for (; shift > 0; shift -= BRANCHING_BASE) {
-            final int skippedElements = array.length - 1;
+            final int skippedElements = obj().lengthOf(array) - 1;
             if (skippedElements != digit(offset, shift)) {
                 break;
             }
-            array = getAt(array, skippedElements);
+            array = obj().getAt(array, skippedElements);
```

Comment:
I could `BitMappedTrie<T> extends ObjectArrayType` to get rid of the `obj()`s, the end result would be cleaner, but it would be kinda' hacky :/